Userland/Libraries/LibJS/Runtime/ArrayBuffer.h

```diff
@@ -34,6 +34,7 @@ class ArrayBuffer : public Object {
     virtual ~ArrayBuffer() override;
 
     size_t byte_length() const { return buffer_impl().size(); }
+    Optional<size_t> max_byte_length() const { return m_max_byte_length; };
```

Comment:
This can probably just return a `size_t` and `VERIFY(m_max_byte_length.has_value())` since it should only be called if `is_resizable_array_buffer()` is true right?

This is not a necessary change just makes the methods slightly nicer to use.

Reply:
Agreed. Also avoids a copy of the optional here.

Reply:
Actually, I don't think this would work - `ArrayBufferPrototype::resize` checks `has_value()`. Let's at least return a `const` reference instead of a copy.

Reply:
Even if it wasn't used for checking `has_value()`, having `max_byte_length` verify that is has a value feels wrong to me. The callsite is responsible for the invariant of the value being present so it should also be responsible for what happens if it's not present.

I'm not sure that the copy of `Optional` here is any more expensive than a reference as it should just be `sizeof(size_t) + sizeof(bool)` plus any padding and the `Optional` copy constructor for trivial types is `default`. That said I have no preference either way and I am significantly less familiar with C++ and the Serenity codebase than either of you so I'll change it to return a `const` reference as requested :)

Reply:
As there is `.is_resizable_array_buffer()` I've removed the location that was calling `.max_byte_length().has_value()` and made `max_byte_length()` verify that it has value and just return the `size_t`. I'm not the fondest of it directly panicking if called incorrectly but I'd rather do what was requested :)